src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -65,6 +72,11 @@ public static void SeekReaderBackwardToMatchByteSignature(BinaryReader reader, b
                 throw new ArgumentNullException(nameof(reader));
             }
 
+            if (byteSignature == null || byteSignature.Length == 0)
+            {
+                throw new ArgumentNullException(nameof(byteSignature));
```

Comment:
throw an appropriate exception for when byteSignature.Length==0 instead of null exception

Reply:
@rohit21agrawal is `InvalidDataException` an appropriate exception? or is it better to have a more generic `ArgumentException`?

Reply:
InvalidDataException sounds correct